server/ntb/commands/fix_events_moment_timezone_2023.py

```diff
@@ -0,0 +1,57 @@
+import superdesk
+
+from datetime import datetime, timedelta
+from flask import current_app as app
+
+
+class FixEventsCommand(superdesk.Command):
+    """Fix events affected by moment timezone issue
+
+    which were saved during summer time using the offset of winter time (or vice versa in Australia).
+    """
+
+    option_list = [
+        superdesk.Option("--start", "-s", default="2023-03-26T01:00:00+00:00"),
+        superdesk.Option("--end", "-e", default="2023-10-29T01:00:00+00:00"),
+        superdesk.Option("--updated", "-u", default="2023-03-20T17:00:00+00:00"),
+        superdesk.Option("--offset", "-o", default="-1"),
+    ]
+
+    def run(self, start, end, updated, offset):
+        start = datetime.fromisoformat(start)
+        end = datetime.fromisoformat(end)
+        updated = datetime.fromisoformat(updated)
+        offset = timedelta(hours=int(offset))
+
+        app.logger.info(
+            "Updating events from %s to %s not updated after %s", start, end, updated
+        )
+
+        query = {
+            "dates.start": {
+                "$gte": start,
+                "$lt": end,
+            },
+            "dates.end": {"$gte": start, "$lt": end},
+            "_updated": {
+                "$lt": updated
+            },  # ignore updated after the instance update, assuming the updates were to fix the dates
+            "ingest_provider": {"$exists": 0},  # ignore ingested events
```

Comment:
This should be fine for NTB, but might be good to skip ingested Events based on a `superdesk.Option`

Reply:
yep in general events which were ingested and not edited should be fine, but I did it kind of simple and expect to adapt it for each instance as needed